Scripts/Source/minai_Sex.psc

```diff
@@ -663,96 +663,110 @@ EndFunction
+          playerInvolved = true
+        EndIf
+        string actorName = Main.GetActorName(actors[i])
+        int excitement = OActor.GetExcitement(actors[i]) as int
+        int orgasmcount = OActor.GetTimesClimaxed(actors[i])
+        if excitement >= 80
+          Main.RegisterEvent(actorName + " is approaching orgasm number " + (orgasmcount+1) + ".")
+        EndIf
+      Endwhile
+      if (playerInvolved)
+        AIFF.ChillOut()
       EndIf
-      string actorName = main.GetActorName(actors[i])
-      int excitement = OActor.GetExcitement(actors[i]) as int
-      int orgasmcount = OActor.GetTimesClimaxed(actors[i])
-      if excitement > 80
-        Main.RegisterEvent(actorName + " is close to orgasm number " + (orgasmcount+1) + ".")
+      Main.Info("Ostim Scene changed to: " + sceneId)
+      Main.Debug("Name: "+sceneId+" isRunning:"+isRunning+" Actions: "+actionTypes.Length+" Actors: "+actorString)
+      string tags
+      int j = actionTypes.Length
+      if j > 0
+        while (j > 0)
+          j -= 1
+          tags = tags+actionTypes[j]+", "
+        Endwhile
       EndIf
-    Endwhile
-    
-    if (playerInvolved)
-      AIFF.ChillOut()
-    EndIf
-    main.RegisterEvent(""+sceneName+" id:"+sceneId+" isRunning:"+isRunning+" Actors:"+actorString,"info_sexscene")
-    Main.Info("Ostim Scene changed")
-
-  elseif (eventName=="ostim_actor_orgasm")    
-    Actor OrgasmedActor = Sender as Actor
-    string sceneName=OThread.GetScene(ostimTid);
-    bool isRunning=OThread.IsRunning(ostimTid);
-    Actor[] actors =  OThread.GetActors(ostimTid);
-    main.RegisterEvent(Main.GetActorName(OrgasmedActor) + " had an Orgasm")
-    DirtyTalk(actors, "ohh... yes.")
-    Main.Info("Ostim Actor orgasm")
-
-  elseif (eventName=="ostim_thread_end")    
-    string sceneName=OThread.GetScene(ostimTid);
-    bool isRunning=OThread.IsRunning(ostimTid);
-    Actor[] actors =  OThread.GetActors(ostimTid);
-    string actorString;
-    int i = actors.Length
-    bool playerInvolved=false
-    while(i > 0)
-      i -= 1
-      actorString=actorString+Main.GetActorName(actors[i])+",";
-      if (actors[i] == playerRef) 
-        playerInvolved=true;
+      int k = sceneTags.Length
+      if k > 0
+        while (k > 0)
+          k -= 1
+          tags = tags+sceneTags[k]+", "
+        Endwhile        
       EndIf
-    endwhile
+      Main.RegisterEvent(actorString + "changed the action to " + tags + ".")
+    else
+      Main.Debug("OStim scene change failed")
+    EndIf
     
-    if (playerInvolved)
-      AIFF.ChillOut()
+  elseif (eventName == "ostim_thread_speedchanged")
+    bool isRunning = OThread.IsRunning(ostimTid)
+    Actor[] actors = OThread.GetActors(ostimTid)
+    if isRunning
+      int sceneSpeed = OThread.GetSpeed(ostimTid)
+      int defaultSpeed = OMetadata.GetDefaultSpeed(ostimTid)
+      if sceneSpeed > defaultSpeed
+        DirtyTalk(actors, "Fuck, yes! I like when you go faster like this.")
```

Comment:
Is DirtyTalk here going to be really spammy? Since speed change is triggered by the LLM. This could lead to a loop.

Reply:
Not so far, dirty talk is throttled anyway.